File: tabla.py
while True:
    numero = int(input('Introduce el número: '))
    if numero < 0 or numero > 10:
        print('El número debe estar comprendido entre 0 y 10')
    else:
        break

contador = -1
while contador < 10:
    contador += 1
    if contador % 2 == 1:
        continue
    print(numero, 'x', contador, '=', numero * contador)

File: java/Tabla.java
import java.util.InputMismatchException;
import java.util.NoSuchElementException;
import java.util.Scanner;

/**
 * Tabla
 */
public class Tabla {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numero;

        for (;;) {
            try {
                System.out.print("Tabla a mostrar: ");
                numero = scanner.nextInt();
                if (numero >= 0 && numero <= 10) {
                    break;
                }
                System.err.println("El número debe estar comprendido entre 0 y 10.");                
            } catch (InputMismatchException e) {
                System.err.println("Debe introducir un número entero.");
                scanner.nextLine();
            } catch (NoSuchElementException e) {
                return;
            }
        }

        for (int i = 0; i <= 10; i++) {
            System.out.print(numero);
            System.out.print(" x ");
            System.out.print(i);
            System.out.print(" = ");
            System.out.println(numero * i);
        }
    }
}